src/functions.php

```diff
@@ -42,6 +43,32 @@
 use function reset;
 use function substr;
 
+/**
+ * Check whether all servers support executing a write stage on a secondary.
+ *
+ * @internal
+ * @param Server[] $servers
+ */
+function all_servers_support_write_stage_on_secondary(array $servers): bool
+{
+    /* Write stages on secondaries are technically supported by FCV 4.4, but the
+     * CRUD spec requires all 5.0+ servers since FCV is not tracked by SDAM. */
+    static $wireVersionForWriteStageOnSecondary = 13;
+
+    foreach ($servers as $server) {
+        // We can assume that load balancers only front 5.0+ servers
+        if ($server->getType() === Server::TYPE_LOAD_BALANCER) {
+            continue;
+        }
```

Comment:
I realize that if there's a load balancer it will be the only server in the topology, so this could also just `return true`. I could also have moved this logic above the call to `all_servers_support_write_stage_on_secondary`, but I was happier to contain this check within the function.